### ISSUE TITLE
finetune hacnn

### ISSUE THREAD

**Issue by chunniunai220ml** [KaiyangZhou/deep-person-reid#95]
Hi, I want to finetune a model by hacnn_market_xent.pth.tar, downloaded from model_zoo . command as follows:
sudo python train_imgreid_xent.py  -s market1501 -t market1501  --optim adam --lr 0.000001 --max-epoch 1 --stepsize 20 40 --train-batch 32 --test-batch 100 --save-dir log/resnet50-xent-market1501 --gpu-devices 0 --load-weights ../../reid_model/hacnn_duke_xent/hacnn_duke_xent.pth.tar --fixbase-epoch 1 --arch hacnn  --height 160 --width 64  --open-layers 'SpatialAttn'
I am confused about  how to set open-layers for hacnn .
and I change the code in https://github.com/KaiyangZhou/deep-person-reid/blob/074a10e24e87eb3cca2d9d697af36f4d29ce1ca2/torchreid/models/hacnn.py#L114
w = x.size(2) 
w=w*2
h = x.size(3) 
h=h*2
x = F.upsample(x, (w,h), mode='bilinear', align_corners=False)
when I convert the hacnn.pth to tensorflow.pb model, the original code (x.size(2) * 2,x.size(3) * 2) can  not recognize as constant leading to convert failed.  I don't know if this change will work. can you give me some advices?

**Comment by chunniunai220ml** [comment]
when I set --open_layer classifier_local ,the loss~6.7, is normally for --load-weights hacnn_duke_xent.pth.tar（you provids in model_zoo ）
![image](https://user-images.githubusercontent.com/17668109/50501797-d9ae6280-0a95-11e9-8ec8-d1522c57d573.png)

**Comment by ciwei123** [comment]
@chunniunai220ml @KaiyangZhou Hi, Where I can find the trained model which has the best performance? I want to quote directly. Thank you very much!

**Comment by KaiyangZhou** [comment]
@chunniunai220ml `--open-layers` should take as input the attribute names of a model, which are specified in `__init__()`, i.e. the immediate children modules, see [this](https://pytorch.org/docs/stable/nn.html#torch.nn.Module.named_children). If you need to finetune some intermediate layers, you need to change `named_children()` to `named_modules()` in [torchtools.py](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/utils/torchtools.py#L58), and specify the absolute attribute names, such as `--open-layers layer1.0.conv layer1.1.conv`. Doing this should work.

**Comment by doublepoints** [comment]
@chunniunai220ml 
Excuse me. I am trying to convert the hacnn model into onnx or tensorflow model. However, I can do it successfully. Did you succeed in converting it ?  And can  you give me some advice?  Thank you very much.
